Mage/src/main/java/mage/abilities/common/BlocksOrBlockedAttachedTriggeredAbility.java

```diff
@@ -2,41 +2,65 @@
 
 import mage.abilities.TriggeredAbilityImpl;
 import mage.abilities.effects.Effect;
+import mage.constants.AttachmentType;
 import mage.constants.Zone;
 import mage.game.Game;
 import mage.game.events.GameEvent;
 import mage.game.permanent.Permanent;
 import mage.target.targetpointer.FixedTarget;
 
 /**
- * @author awjackson
+ * @author awjackson & TiagoMDG
  */
 public class BlocksOrBlockedAttachedTriggeredAbility extends TriggeredAbilityImpl {
 
-    public BlocksOrBlockedAttachedTriggeredAbility(Effect effect) {
-        this(effect, false);
+    private final AttachmentType attachmentType;
+    private final boolean selfTarget;
+
+    public BlocksOrBlockedAttachedTriggeredAbility(Effect effect, AttachmentType attachmentType, boolean selfTarget) {
+        this(effect, attachmentType, selfTarget, false);
     }
-    public BlocksOrBlockedAttachedTriggeredAbility(Effect effect, boolean optional) {
+
+    public BlocksOrBlockedAttachedTriggeredAbility(Effect effect, AttachmentType attachmentType, boolean selfTarget, boolean optional) {
         super(Zone.BATTLEFIELD, effect, optional);
-        setTriggerPhrase("Whenever enchanted creature blocks or becomes blocked, ");
+        setTriggerPhrase("Whenever " + attachmentType.verb().toLowerCase() + " creature blocks or becomes blocked, ");
+        this.attachmentType = attachmentType;
+        this.selfTarget = selfTarget;
     }
 
     public BlocksOrBlockedAttachedTriggeredAbility(final BlocksOrBlockedAttachedTriggeredAbility ability) {
         super(ability);
+        this.attachmentType = ability.attachmentType;
+        this.selfTarget = ability.selfTarget;
     }
 
     @Override
     public boolean checkEventType(GameEvent event, Game game) {
-        return event.getType() == GameEvent.EventType.CREATURE_BLOCKS
-                || event.getType() == GameEvent.EventType.CREATURE_BLOCKED;
+        return event.getType() == GameEvent.EventType.BLOCKER_DECLARED;
     }
 
     @Override
     public boolean checkTrigger(GameEvent event, Game game) {
         Permanent creature = game.getPermanent(event.getTargetId());
+        Permanent equipment = game.getPermanent(sourceId);
```

Comment:
Let's call this "attachment" instead of "equipment", and make sure to null check it as well